iam.tf

```
@@ -0,0 +1,100 @@
+locals {
+  create_role      = local.enabled && (var.existing_iam_role_arn == null || var.existing_iam_role_arn == "")
+  role_arn         = local.create_role ? aws_iam_role.default[0].arn : var.existing_iam_role_arn
```

I've been tripped up by this a few times. Although it might seem like there should always be a `default[0]` if we are creating one, there might not be one when `enabled = false` (particularly when previously deployed with `enabled = true`) and/or during `destroy`. I think it is best/safest to always use `one()`.
```suggestion
  role_arn         = local.create_role ? one(aws_iam_role.default[*].arn) : var.existing_iam_role_arn
```